app/Main.hs

```diff
@@ -1,16 +1,26 @@
-{-# LANGUAGE FlexibleContexts #-}
 module Main where
 
 import           Control.Monad.State.Lazy
 import           Data.Char
+import           Data.Maybe
 import           System.Environment
 
-initialCellState = replicate 30000 (0 :: Int)
-
 type CellState = [Int]
 
--- MoveState :: Int (Position) -> Parsed Commands -> CellState
-type MoveState = (Int, [Command], CellState)
+data MoveState
```

Comment:
Cool, was looking into using records but just settled for tuples as it was easier